zerver/context_processors.py

```diff
@@ -198,6 +198,9 @@ def login_context(request: HttpRequest) -> Dict[str, Any]:
         "require_email_format_usernames": require_email_format_usernames(realm),
         "password_auth_enabled": password_auth_enabled(realm),
         "two_factor_authentication_enabled": settings.TWO_FACTOR_AUTHENTICATION_ENABLED,
+        # We offer web public access if the realm has actual web
+        # public streams configured.
+        "offer_web_public_access": realm.has_web_public_streams(),
```

Comment:
This should ideally be [allow_web_public_streams_access](https://github.com/zulip/zulip/pull/18532/commits/8feb78030e58d146f643484fbe10b31c2c4bf8f2#diff-c210d4a13c8c2d0d481c3755996fb1dfe0dadd76cc1bcba6e817ee9bfa29c7e0R892) after 8feb78030e58d146f643484fbe10b31c2c4bf8f2 gets merged.

Reply:
Yeah, changing that sounds good.

Reply:
I will adjust it in #18532 after this gets merged.